topicExtractor.py

```diff
@@ -0,0 +1,169 @@
+from config import OpenAIBot, LangChainBot
+from utils import dataLoader, dataSaver, printAndLog
+from bertopic import BERTopic
+from keybert import KeyBERT
+from sklearn.feature_extraction.text import CountVectorizer
+from langchain_community.callbacks import get_openai_callback
+
+
+def retrieveTopics(
+    videoToUse, transcriptToUse, config, overwrite=False, saveNameAppend=""
+):
+    """
+    Retrieves or generates topic model and data.
+
+    Args:
+        videoToUse (str): Path to the video file.
+        transcriptToUse (str): Path to the transcript file.
+        config (dict): Configuration settings.
+        overwrite (bool, optional): Whether to overwrite existing data. Defaults to False.
+        saveNameAppend (str, optional): String to append to the save name. Defaults to "".
+
+    Returns:
+        tuple: A tuple containing the topics over time and the topic model.
+    """
+
+    if not overwrite and (
+        (topicModel := dataLoader(config, "topicModel", videoToUse, saveNameAppend))
+        is not False
+        and (
+            topicsOverTime := dataLoader(
+                config, "topicsOverTime", videoToUse, saveNameAppend
```

Comment:
Can you explain what is the difference b/w `topicModel` and `topicsOverTime`?I think I am not clean by those means. I feel it's good idea to explain here as well.

Reply:
I will add an explanation in the ReadMe if that works.
topicModel is the actual Model itself, while topicsOverTime are the extracted topics from said model. 
There is a compute cost to getting the extracted topics, and for debugging and visualization, there is a need to keep the topic Model itself so we can call it for any other uses. Hence they are separate entities.

Reply:
I feel just write it in the function docstring here since this keywords are first mentioned as part of this code.

Reply:
I have made changes in the new [Commit](https://github.com/tl-its-umich-edu/annoto-gai/pull/14/commits/12992ec3fadf5a4771519490c1f580d4f442db7b). This puts them together into a single class, so it better reflects that they are part of the same object and connected to each other. It should make it clearer, if not, I will elaborate further in the `TopicModeller`'s docstring.